x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,206 @@
+	PreviousMoniker string         `json:"previous_moniker"`
+	NewMoniker      string         `json:"new_moniker"`
+	Name            string         `json:"name,omitempty"`
+	Surname         string         `json:"surname,omitempty"`
+	Bio             string         `json:"bio,omitempty"`
+	Pictures        *Pictures      `json:"pictures,omitempty"`
+	Creator         sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgEditProfile is a constructor function for MsgEditProfile
+func NewMsgEditProfile(previousMoniker string, newMoniker string, name string, surname string, bio string, pictures *Pictures,
+	creator sdk.AccAddress) MsgEditProfile {
+	return MsgEditProfile{
+		PreviousMoniker: previousMoniker,
+		NewMoniker:      newMoniker,
+		Name:            name,
+		Surname:         surname,
+		Bio:             bio,
+		Pictures:        pictures,
+		Creator:         creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgEditProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgEditProfile) Type() string { return ActionEditProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgEditProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(strings.TrimSpace(msg.PreviousMoniker)) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Profile previous moniker cannot be blank or empty")
+	}
+
+	if len(msg.PreviousMoniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile previous moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.NewMoniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile new moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Bio) > MaxBioLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile biography cannot exceed %d characters", MaxBioLength))
+	}
+
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg MsgEditProfile) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgEditProfile) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Creator}
+}
+
+// ----------------------
+// --- MsgDeleteProfile
+// ----------------------
+
+// MsgDeleteProfile defines a DeleteProfile message
+type MsgDeleteProfile struct {
+	Moniker string         `json:"moniker"`
```

Comment:
Probably this is not required, we should be able to delete the profile of a user only based on his address.